src/App.js

```diff
@@ -19,8 +21,12 @@ export class App extends Component {
       mode = 'default',
       lang = 'en',
     } = Qs.parse(window.location.search, { ignoreQueryPrefix: true });
-    const { i18n } = this.props;
-    i18n.changeLanguage(lang);
+    const { i18n, defaultLanguage } = this.props;
```

Comment:
We agreed that the below bit should go in the `store` when you configured it. You should always get the language from the reducer.